python_ta/debug/accumulation_table.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+import inspect
+from typing import Union
+
+import astroid
+import tabulate
+
+
+class AccumulationTable:
+    """
+    Class used as a form of print debugging to analyze different loop and
+    accumulation variables during each iteration in a for loop
+
+    Private instance attributes:
+        _num_iterations: the number of for loop iterations
+        _accumulation_names: the variable names of the different accumulation variables
+        _loop_accumulation_values: a mapping between the accumulation variables and their value during each iteration
+        _loop_var_name: the name of the loop variable
+        _loop_var_val: the value of the loop variable during each iteration
+
+    """
+
+    _num_iterations: int
+    _accumulation_names: list[str]
+    _loop_accumulation_values: dict[str, list]
+    _loop_var_name: str
+    _loop_var_val: list[Union[int, str]]
+
+    def __init__(self, accumulation_names: list) -> None:
+        self._accumulation_names = accumulation_names
+        self._num_iterations = 0
+        self._loop_var_val = ["N/A"]
+
+    def add_iteration(self, val: list, var) -> None:
+        """Add the values of the accumulator and loop variables of an iteration"""
+        self._loop_var_val.append(var)
+        for i in range(len(self._accumulation_names)):
+            self._loop_accumulation_values[self._accumulation_names[i]].append(val[i])
+        self._num_iterations += 1
+
+    def _create_iteration_dict(self) -> dict:
+        """Use the values of each iteration and return a dictionary that maps each accumulator
+        and loop variable to its respective value during each iteration
+        """
+        table_dict = {
+            "iterations": list(range(self._num_iterations + 1)),
+            "loop variable (" + self._loop_var_name + ")": self._loop_var_val,
+            **self._loop_accumulation_values,
+        }
+        return table_dict
+
+    def _tabulate_data(self) -> None:
+        """Print the values of the accumulator and loop variables into a table"""
+        print(
+            tabulate.tabulate(
+                self._create_iteration_dict(), headers="keys", colalign=("left", "left", "left")
+            )
+        )
+
+    def _no_of_indents(self, with_line: str) -> int:
```

Comment:
This is definitely the right idea, but I think needs to be a bit more general. The issue is that Python allows different sizes for indentation (e.g. 2 spaces instead of 4) and allows tabs (`'\t'`) as well.

So you can improve your approach to just count the number of whitespace characters rather than the number of indents. The same logic you're using in `__enter__` will work, you just need to compare the the number of whitespace characters rather than the number of indentations